src/app/components/AutoComplete.svelte

```diff
@@ -1,20 +1,32 @@
 <script lang="ts">
-  // TODO: the following interface should be removed once #33 is done
-  interface dataObj {
+  import { onMount } from "svelte";
+  import worker from "../spawn-worker";
+  interface KeyboardDataWithTime {
+    /**
+     * Store bcp47Tag as the primarykey
+     */
     bcp47Tag: string;
+    /**
+     * Keyboard Language of the bcp47Tag
+     */
     language: string;
+    /**
+     * Timestamp of last updated time
+     */
+    timestamp: Date;
   }
```

Comment:
Can this interface be placed in `src/common/types.ts`? That way, it's not duplicated all over the place. Note that TypeScript has "structural subtyping" so an interface doesn't need to _explicilty_ `extend` a different interface to be compatible.

Reply:
I moved the interface to `src/common/types.ts` but Svelte is complaining that the interface is not being exported from the file although I have put an export there 😭. Is this because the component file is being bundled before the worker in which it doesn't not recognize the interface is in the file?

Reply:
```suggestion
  import type { KeyboardDataWithTime } from "@common/types";
```

We ran into this issue before ;_; you need to `import type` or else svelte-preprocess will try to actually import some sort of concrete object from `@common/types`. See:

https://github.com/sveltejs/svelte-preprocess/blob/main/docs/preprocessing.md#typescript---limitations